src/Implementation/ExecutableFactory.cs

```diff
@@ -31,10 +31,7 @@ public IProgram CreateInitial(IProgramPool programPool, IModule module, Options
             var stack = new StackProxy(space, memory, CreateStack(module, options));
             var system = new SystemProxy(space, memory, CreateSystem(module));
 
-            var state = new State(programPool, module, space,
-                globals, memory, stack, system);
-
-            return new Program(() => state);
+            return new State(_ => { }, module, space, globals, memory, stack, system);
```

Comment:
Do you think we should have a `State.CreateInitial` static method?

Reply:
Nah.